accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,277 @@
+At the time of writing, registering a CaaS Platform node to Uyuni/SUSE Manager has to be manually performed by the user with:
+  * Bootstrap script
+  * UI: Systems > Bootstrapping
+  * `bootstrap` XMLRPC call
+  * Not usual but still possible and hackier: mass bootstrap via salt-ssh and a user-generated roster file
+
+## Assumptions
+
+We assume that:
+* The user is registering the CaaS Platform nodes as Salt clients
+* The registration of every CaaS Platform node to Uyuni/SUSE Manager is already completed by the user using the aforementioned methods
+* The user has already created an activation key and associated it with the onboarding. The CaaS Platform nodes are registered correctly and with the proper CaaS Platform channel(s) assigned.
+
+NOTE: It is outside of this RFC to discuss automatic registration methods of the CaaS Platform nodes.
+
+## Technical description of the problem
+
+### The forbidden actions
+
+[`skuba-update`](https://documentation.suse.com/suse-caasp/4/single-html/caasp-admin/#_base_os_updates) systemd timer will take care of [patching](https://github.com/SUSE/skuba/blob/master/skuba-update/skuba_update/skuba_update.py#L301-L305) all installed packages locally to every node of the cluster and optionally reboot the node if `zypper` signals that a reboot is required.
+
+Uyuni/SUSE Manager should not offer the possibility to reboot a registered CaaS Platform node.
+
+In general, patch and package upgrade/removal are safe operations for a CaaS Platform cluster: on the next trigger of `skuba-update`, if there are packages that required a reboot, `skuba-update` will orchestrate the reboot.
+
+Package upgrade/removal and patch apply is not a problem if issued from Uyuni/SUSE Manager, unless the involved packages of the action are one of the following:
+
+- `kubernetes-kubeadm`
+- `kubernetes-kubelet`
+- `kubernetes-client`
+- `cri-o`
+- `cni-plugins`
+
+NOTE: [installing NEW (not already installed) packages is allowed](https://documentation.suse.com/suse-caasp/4/single-html/caasp-admin/#_existing_cluster).
+
+At the time of the writing, Uyuni/SUSE Manager or Salt cannot forbid the package operations based on the target packages (there is no such kind of granularity).
+
+### Actions to forbid
+
+Decision criteria:
+
+* Either Uyuni/SUSE Manager allows patching and package upgrade/removal or it does not allow those operations.
+* Additionally, the list of the "forbidden" packages above may vary with time and with CaaS Platform upgrades. Currently, there is no programmatic way to gather a list of those packages from a CaaS Platform cluster. What if Uyuni/SUSE Manager hardcodes that list of packages and the list changes? An update of Uyuni/SUSE Manager would be required.
+
+For the reasons above, Uyuni/SUSE Manager should not interfere with these actions to avoid breaking the cluster. The following actions should not be executed from Uyuni/SUSE Manager:
+
+- System reboot
```

Comment:
For completeness' sake I'd add power management actions here, and SP migration

Reply:
Right, I will add it